### PR TITLE
WIP key store set up

### PR DESCRIPTION
**Pull request by cavster**
added in session filter and caching api

Changed desscibesion

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala [hunk 1]
```diff
@@ -16,12 +16,13 @@
 
 package uk.gov.hmrc.helptosavefrontend
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala]: ide  warning : package name doesnt correspond to directory structure, ...`config`

File: app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveController.scala [hunk 2]
```diff
@@ -16,7 +16,9 @@
 
 package uk.gov.hmrc.helptosavefrontend.controllers
 
-import play.api.mvc.{Action, AnyContent, Request, Result}
+import java.util.UUID
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveController.scala]: unused import UUID

File: app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala [hunk 3]
```diff
@@ -70,3 +78,26 @@ object AuditFilter extends FrontendAuditFilter with RunMode with AppName with Mi
...
+  * Checks if the request has cookie with a HTS session ID in it - if not, the filter creates one and
+  * performs `whenNoSession` with the new session ID in it
+  */
+@Singleton
+class SessionFilter[A](whenNoSession: => Result)(implicit app: Application) extends EssentialFilter {
+
+  implicit val ex = app.injector.instanceOf[ExecutionContext]
+  val sessionIdKey = app.configuration.underlying.getString("microservice.services.keystore.session-key")
+
+  private def createHtsCookie() = Cookie(name = sessionIdKey, value = s"hts-session-${UUID.randomUUID}")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala]: is it possible to set cookie life time here ?

Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala]: How secure is `UUID`? What are the chances of creating duplicate IDs?

Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendGlobal.scala]: The `hts-session` prefix in the cookie value seems redundant given that the name already qualifies it

File: app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveController.scala (hunk 2 above)
Comment (cavster) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveController.scala]: It's being used on line 92 ;)

**Pull request merged**